### PR TITLE
CI and Build

### PR DESCRIPTION
**Pull request by gbenhaim**
- Add Github action for running unit tests a linting.
- Configure the Dockerfile.
- Downgrade to golang 1.21.9 since UBI images doesn't offer golang 1.22
- Fix code issues the linter complained about.

### CODE REVIEW

File: Dockerfile [hunk 1]
```diff
@@ -23,11 +22,10 @@ COPY internal/controller/ internal/controller/
 # by leaving it empty we can ensure that the container and binary shipped on it will have the same platform.
 RUN CGO_ENABLED=0 GOOS=${TARGETOS:-linux} GOARCH=${TARGETARCH} go build -a -o manager cmd/main.go
 
-# Use distroless as minimal base image to package the manager binary
-# Refer to https://github.com/GoogleContainerTools/distroless for more details
-FROM gcr.io/distroless/static:nonroot
-WORKDIR /
-COPY --from=builder /workspace/manager .
+FROM registry.access.redhat.com/ubi8/ubi-minimal:8.9-1029
```
Comment (yftacherzog) [inline Dockerfile:25]: What's the reason for using ubi8 and not 9?

Comment (gbenhaim) [reply Dockerfile:25]: No good reason. We can use rhel 9.

Comment (gbenhaim) [reply Dockerfile]: I changed the images to use rhel 9

**Comment by yftacherzog** [comment]
I see no workflows running. Do you know how come?

**Comment by gbenhaim** [comment]
@yftacherzog I think it starts to run after your merge it for the first time.

**Comment by yftacherzog** [comment]
> @yftacherzog I think it starts to run after your merge it for the first time.

ok. I thought this is not the case if you're a maintainer.

**Pull request merged**